api/src/main/java/org/xnio/XnioWorker.java

```diff
@@ -874,6 +877,12 @@ protected final int getWorkerQueueSize() {
         return taskQueue.size();
     }
 
+    public abstract XnioWorkerMXBean getMetrics();
+
+    protected abstract Closeable registerServerMetrics(XnioServerMXBean metrics);
```

Comment:
If this returns a "Closeable" then the "unregister" below isn't needed.  Also this could be final not abstract, and just directly add the mxbean to the set on Xnio.